environment/src/main/java/com/sequenceiq/environment/proxy/service/ProxyConfigService.java

```diff
@@ -0,0 +1,91 @@
+package com.sequenceiq.environment.proxy.service;
+
+import static com.sequenceiq.cloudbreak.common.exception.NotFoundException.notFound;
+
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.Sets;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.cloudbreak.logger.MDCBuilder;
+import com.sequenceiq.environment.proxy.domain.ProxyConfig;
+import com.sequenceiq.environment.proxy.repository.ProxyConfigRepository;
+
+@Service
+public class ProxyConfigService  {
+
+    private final ProxyConfigRepository proxyConfigRepository;
+
+    public ProxyConfigService(ProxyConfigRepository proxyConfigRepository) {
+        this.proxyConfigRepository = proxyConfigRepository;
+    }
+
+    public ProxyConfig get(Long id) {
+        return proxyConfigRepository.findById(id).orElseThrow(notFound("Proxy configuration", id));
+    }
+
+    public ProxyConfig deleteInAccount(String name, String accountId) {
+        ProxyConfig proxyConfig = proxyConfigRepository.findByNameInAccountId(name, accountId)
+                .orElseThrow(notFound("Proxy config with name:", name));
+        MDCBuilder.buildMdcContext(proxyConfig);
+        proxyConfigRepository.delete(proxyConfig);
+        return proxyConfig;
+    }
+
+    public Set<ProxyConfig> deleteMultipleInAccount(Set<String> names, String accountId) {
+        Set<ProxyConfig> toBeDeleted = getByNamesForAccountId(names, accountId);
+        return toBeDeleted.stream()
+                .map(credential -> deleteInAccount(credential.getName(), accountId))
+                .collect(Collectors.toSet());
+    }
+
+    private Set<ProxyConfig> getByNamesForAccountId(Set<String> names, String accountId) {
+        Set<ProxyConfig> results = proxyConfigRepository.findByNamesInAccount(names, accountId);
+        Set<String> notFound = Sets.difference(names,
+                results.stream().map(p -> p.getName()).collect(Collectors.toSet()));
```

Comment:
p -> p.getName() could be replaced with method reference: ProxyConfig::getName

Reply:
ok